geom/dcel_overlay.go

```diff
@@ -16,70 +16,21 @@ func createOverlay(a, b Geometry) (*doublyConnectedEdgeList, error) {
 		return nil, wrap(err, "re-noding")
 	}
 
-	interactionPoints := findInteractionPoints([]Geometry{a, b, ghosts.AsGeometry()})
+	interactions := findInteractionPoints([]Geometry{a, b, ghosts.AsGeometry()})
 
-	dcelA := newDCELFromGeometry(a, ghosts, operandA, interactionPoints)
-	dcelB := newDCELFromGeometry(b, ghosts, operandB, interactionPoints)
+	dcel := newDCEL()
+	dcel.addVertices(interactions)
+	dcel.addGhosts(ghosts, interactions)
+	dcel.addGeometry(a, operandA, interactions)
+	dcel.addGeometry(b, operandB, interactions)
 
-	dcelA.overlay(dcelB)
-	return dcelA, nil
-}
-
-func (d *doublyConnectedEdgeList) overlay(other *doublyConnectedEdgeList) {
-	d.overlayVertices(other)
-	d.overlayEdges(other)
-	d.fixVertices()
-	d.reAssignFaces()
-	d.populateInSetLabels()
-}
-
-func (d *doublyConnectedEdgeList) overlayVertices(other *doublyConnectedEdgeList) {
-	for xy, otherVert := range other.vertices {
-		vert, ok := d.vertices[xy]
-		if ok {
-			mergeBools(&vert.src, otherVert.src)
-			mergeBools(&vert.inSet, otherVert.inSet)
-			mergeLocations(&vert.locations, otherVert.locations)
-		} else {
-			d.vertices[xy] = otherVert
-		}
-	}
-	for _, e := range other.halfEdges {
-		if existing, ok := d.vertices[e.origin.coords]; ok {
-			e.origin = existing
-		} else {
-			d.vertices[e.origin.coords] = e.origin
-		}
-	}
-}
+	dcel.fixVertices()
+	dcel.reAssignFaces()
+	dcel.populateInSetLabels()
 
-func (d *doublyConnectedEdgeList) overlayEdges(other *doublyConnectedEdgeList) {
-	// Clear incidents lists, since we're going to re-compute them.
-	for _, vert := range d.vertices {
-		vert.incidents = nil
-	}
-	for _, vert := range other.vertices {
-		vert.incidents = nil
-	}
+	//dumpDCEL(dcel)
```

Comment:
Did you intend to leave this debugging line in?

Reply:
Woops! Removed in b8c7c72b51fcd2f423707f8227e1b8fceb112c51